File: bender/app/src/main/java/kz/app/bender/di/components/ComponentNet.java
package kz.app.bender.di.components;

import javax.inject.Singleton;

import dagger.Component;
import kz.app.bender.ApplicationMain;
import kz.app.bender.di.DIWrapper;
import kz.app.bender.di.modules.ModuleApp;
import kz.app.bender.mvp.AbstractModel;

@Singleton
@Component(modules = {ModuleApp.class, ModuleNet.class})
public interface ComponentNet {
    void inject(ApplicationMain application);

    void inject(DIWrapper diWrapper);

    void inject(AbstractModel abstractModel);
}

File: Alem/app/src/main/java/kz/alemtime/alem/di/components/ComponentNet.kt
package kz.alemtime.alem.di.components

import dagger.Component
import kz.alemtime.alem.ApplicationMain
import kz.alemtime.alem.di.DIWrapper
import kz.alemtime.alem.di.modules.ModuleApp
import kz.alemtime.alem.di.modules.ModuleNet
import kz.alemtime.alem.mvp.AbstractModel
import javax.inject.Singleton

@Singleton
@Component(modules = [ModuleApp::class, ModuleNet::class])
interface ComponentNet {
    fun inject(application: ApplicationMain)

    fun inject(diWrapper: DIWrapper)

    fun inject(abstractModel: AbstractModel)
}